Mathlib/Algebra/Polynomial/Degree/IsMonicOfDegree.lean

```diff
@@ -0,0 +1,275 @@
+/-
+Copyright (c) 2025 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.Algebra.Polynomial.AlgebraMap
+import Mathlib.Algebra.Polynomial.Monic
+import Mathlib.Tactic.ComputeDegree
+
+/-!
+# Monic polynomials of given degree
+
+This file defines the predicate `Polynomial.IsMonicOfDegree p n` that states that
+the polynomial `p` is monic and has degree `n` (i.e., `p.natDegree = n`.)
+
+We also provide some basic API.
+-/
+
+namespace Polynomial
+
+/-- This says that `p` has `natDegree` `n` and is monic. -/
+def IsMonicOfDegree {R : Type*} [Semiring R] (p : R[X]) (n : ℕ) : Prop :=
```

Comment:
Have you thought about using a structure here?

Reply:
Not really. What would be the benefits of doing that?

Reply:
You get for free `IsMonicOfDegree.monic` and `IsMonicOfDegree.natDegree_eq` (and `@[mk_iff]` should automatically generate the relevant iff lemma).

Reply:
OK; I will change that (have to teach now, though...).